### PR TITLE
Support Thread 1.4 reference builds

### PR DESCRIPTION
**Pull request by suveshpratapa**

### CODE REVIEW

**Comment by canisLupus1313** [comment]
@LuDuda FYI

File: script/make-firmware.bash [hunk 1]
```diff
@@ -275,21 +285,36 @@ build()
...
+    elif [ "${REFERENCE_RELEASE_TYPE}" = "1.3" ]; then
         options=("${build_1_3_options_common[@]}")
 
         case "${platform}" in
             nrf*)
                 options+=("${build_1_3_options_nrf[@]}")
                 platform_repo=ot-nrf528xx
 
-                thread_version=1.2 build_type="USB_trans" build_ot "${options[@]}" "$@"
+                thread_version=1.3 build_type="USB_trans" build_ot "${options[@]}" "$@"
```
Comment (superwhd) [inline script/make-firmware.bash:296]: I think 1.3 and 1.4 are not supported in `build_ot`

Comment (superwhd) [reply script/make-firmware.bash:296]: It seems that the firmware options is changed by this change when `REFERENCE_RELEASE_TYPE` is `1.3`. I wonder if this as expected.

@sunytt What do you think?

Comment (superwhd) [reply script/make-firmware.bash:296]: My mistake. Actually the build options weren't changed when REFERENCE_RELEASE_TYPE is 1.3. The `thread_version` variable doesn't change the build options in `build_ot`.

The change looks reasonable to me.

**Review by superwhd — APPROVED** [review]
LGTM 👍 

It seems the CI is failing due to low CMake version.

**Comment by sunytt** [comment]
`.github/workflows/build.yml` still builds 1.3.1, could you update it to 1.4?

**Comment by superwhd** [comment]
@suveshpratapa It seems bumping the version of ot-commissioner is not a trivial step. There may be more actions needed to fix the CI failure: https://github.com/openthread/ot-reference-release/actions/runs/10164731867/job/28110844136?pr=79.

Can we bump it in another PR?

**Comment by canisLupus1313** [comment]
@suveshpratapa  The build problems of NCS are addressed here https://github.com/openthread/ot-reference-release/pull/82

**Comment by superwhd** [comment]
The ot-commissioner's build failure should be fixed by https://github.com/openthread/ot-commissioner/pull/291.

**Comment by jwhui** [comment]
> @jwhui Thank you, just incorporated into this PR. Now just need to wait for Nordic PR #82 to be merged.

Merged the Nordic PR as well.

**Review by canisLupus1313 — APPROVED** [review]

**Comment by superwhd** [comment]
@suveshpratapa Since there are already some build failures and the PR includes multiple tasks (script change and bumping repos), could you consider dividing the PR into multiple ones? That should make it easier for debugging the build failures.

Maybe you can create a PR for bumping the repos and another PR for script changes.

**Comment by superwhd** [comment]
> The hosted runner: GitHub Actions 403 lost communication with the server. Anything in your workflow that terminates the runner process, starves it for CPU/Memory, or blocks its network access can cause this error.

https://github.com/openthread/ot-reference-release/actions/runs/10216991615?pr=79

**Comment by canisLupus1313** [comment]
@suveshpratapa The NCS 1.4 build fails for you because the reference configs do not exist for 1.4 yet in both Zephyr and NCS. We are working curently on adding 1.4 for now You can leave 1.3.1 ad default in builds for 1.4. It should work ok for You.

**Comment by maciejbaczmanski** [comment]
@suveshpratapa https://github.com/openthread/ot-reference-release/pull/85 should allow building thread 1.4 in NCS

**Comment by maciejbaczmanski** [comment]
> > @suveshpratapa #85 should allow building thread 1.4 in NCS
> 
> @maciejbaczmanski Can you check what could be happening? I rebased to include your merged changes and t's still failing to build NCS for 1.4.

error can be seen here: https://github.com/openthread/ot-reference-release/actions/runs/10404358834

`System.IO.IOException: No space left on device : '/home/runner/runners/2.319.1/_diag/Worker_20240815-131230-utc.log'`

So it seems like a CI issue?

**Comment by jwhui** [comment]
> error can be seen here: https://github.com/openthread/ot-reference-release/actions/runs/10404358834
> 
> `System.IO.IOException: No space left on device : '/home/runner/runners/2.319.1/_diag/Worker_20240815-131230-utc.log'`
> 
> So it seems like a CI issue?

I just submitted https://github.com/openthread/ot-reference-release/pull/87, hopefully it will help.

**Comment by jwhui** [comment]
@suveshpratapa , I just merged #87. Can you rebase this PR and see if it helps?

**Review by maciejbaczmanski — APPROVED** [review]

**Review by superwhd — APPROVED** [review]
LGTM 👍 

The CI finally passed! 🎉

**Review by jwhui — APPROVED** [review]

**Pull request merged**